packages/notification-service/src/firebase.ts

```diff
@@ -139,12 +119,26 @@ export function initializeDb() {
   knownAddressesCache.startListening(database)
 }
 
+export function getRegistration(address: string) {
+  let registration: Registration | undefined
+  registrationsRef.on(
```

Comment:
I believe we should use `once` instead and make this function async.
Otherwise we create a listener while we're only interested in a single read.

Reply:
you're so right on both fronts, thanks for catching this! i have updated